File: Array/zhuqiguang/java/Solution_206.java

public class Solution_206 {
    public ListNode reverseList(ListNode head) {
        ListNode pre = null;
        ListNode cur = head;
        ListNode tmp = null;
        while(cur != null) {
            tmp = cur.next;
            cur.next = pre;
            pre = cur;
            cur = tmp;
        }
        return pre;
    }
}

File: Array/zhuqiguang/c/Solution_206.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *tmp = NULL;
    struct ListNode *pre = NULL;
    struct ListNode *cur = head;
    while(cur != NULL) {
        tmp = cur ->next;
        cur->next = pre;
        pre = cur;
        cur = tmp;
    }
    return pre;
}